tests/check_framework/test_caches.py

```diff
@@ -28,3 +33,52 @@ def test_default_cache_not_included(self):
         Error if 'default' not present in CACHES setting.
         """
         self.assertEqual(check_default_cache_is_configured(None), [E001])
+
+
+class CheckCacheLocationTest(SimpleTestCase):
+    @staticmethod
+    def get_settings(name, cache_path, other_path):
+        return {
+            'CACHES': {
+                'default': {
+                    'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
+                    'LOCATION': cache_path,
+                },
+            },
+            name: [other_path] if name == 'STATICFILES_DIRS' else other_path,
+        }
+
+    def assertWarningWithHint(self, hint):
+        self.assertEqual(check_cache_location_not_exposed(None), [Warning(
+            'Your configuration might expose your cache or lead to corruption of your data.',
+            hint=hint,
+            id='cache.W002',
+        )])
```

Comment:
I think it's more readable without this hook.

Reply:
Yes, thanks for review!